<?xml version="1.0"?>
 <launch>
   <arg name="namespace" default="" />
   <arg name="dataset" default="hmdb51"/>
   <arg name="y_hat_input" default="scores_fw"/>
   <arg name="filenamesux" default="$(arg namespace)"/>
   <node name="trainer" pkg="datasetloader_ros" type="trainer.py" respawn="false" output="screen" ns="$(arg namespace)">
     <param name="glistfile"  value="/mnt/externaldrive/var/datasets/$(arg dataset)/global_pool_train_$(arg filenamesux)" />
     <param name="tot"        value="train" />
     <!-- std_msgs.msg.Float32MultiArray , String, caffe_tsn_ros.msg.ScoreArray  -->
     <!-- <param name="yhat_type"  value="caffe_tsn_ros.msg.ScoreArray" /> -->
     <param name="yhat_type"  value="caffe_tsn_ros.msg.ScoreArray" />

     <param name="y_type"     value="std_msgs.msg.String"/>
     <remap from="y_hat_topic" to="/$(arg y_hat_input)"/>
     <remap from="y_topic" to="/readpathnode/y"/>
     <remap from="done_topic" to="/readpathnode/done"/>
     <param name="synch_appending" type="bool" value="True"/>
     <param name="namespace" value="$(arg namespace)"/>
     <rosparam param="classes">['brush_hair', 'cartwheel', 'catch', 'chew', 'clap', 'climb', 'climb_stairs', 'dive', 'draw_sword', 'dribble', 'drink', 'eat', 'fall_floor', 'fencing', 'flic_flac', 'golf', 'handstand', 'hit', 'hug', 'jump', 'kick','kick_ball', 'kiss', 'laugh', 'pick', 'pour', 'pullup', 'punch', 'push', 'pushup', 'ride_bike', 'ride_horse', 'run', 'shake_hands', 'shoot_ball', 'shoot_bow', 'shoot_gun', 'sit', 'situp', 'smile', 'smoke', 'somesault', 'stand', 'swing_baseball', 'sword', 'sword_exercise', 'talk', 'throw', 'turn', 'walk', 'wave', 'unknown']</rosparam>
     <param name="rate" value="1"/>
   </node>
 </launch>
